src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/VariableFetchAnalyzer.php

```diff
@@ -531,26 +540,231 @@ public static function getGlobalType(string $var_id): Union
+                'REDIRECT_STATUS'      => $non_empty_string_helper,
+                'REQUEST_SCHEME'       => $non_empty_string_helper,
+                'USER'                 => $non_empty_string_helper,
+                // common, misc headers
+                'HTTP_UPGRADE_INSECURE_REQUESTS' => $non_empty_string_helper,
+                'HTTP_X_FORWARDED_PROTO'         => $non_empty_string_helper,
+                'HTTP_CLIENT_IP'                 => $non_empty_string_helper,
+                'HTTP_X_REAL_IP'                 => $non_empty_string_helper,
+                'HTTP_X_FORWARDED_FOR'           => $non_empty_string_helper,
+                'HTTP_CF_CONNECTING_IP'          => $non_empty_string_helper,
+                'HTTP_CF_IPCOUNTRY'              => $non_empty_string_helper,
+                'HTTP_CF_VISITOR'                => $non_empty_string_helper,
+                'HTTP_CDN_LOOP'                  => $non_empty_string_helper,
+                // common, misc browser headers
+                'HTTP_DNT'                => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_DEST'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_USER'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_MODE'     => $non_empty_string_helper,
+                'HTTP_SEC_FETCH_SITE'     => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA_PLATFORM' => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA_MOBILE'   => $non_empty_string_helper,
+                'HTTP_SEC_CH_UA'          => $non_empty_string_helper,
             ]);
+
+            // generic case for all other elements
+            $detailed_type->previous_key_type = Type::getNonEmptyString();
+            $detailed_type->previous_value_type = Type::getString();
+
+            return new Union([$detailed_type]);
         }
 
-        if (self::isSuperGlobal($var_id)) {
-            $type = Type::getArray();
-            if ($var_id === '$_SESSION') {
-                $type->possibly_undefined = true;
+        if ($var_id === '$_FILES') {
+            $values = [
+                'name' => new Union([
+                    new TString(),
+                    new TNonEmptyList(Type::getString()),
+                ]),
+                'type' => new Union([
+                    new TString(),
+                    new TNonEmptyList(Type::getString()),
+                ]),
+                'size' => new Union([
+                    new TInt(),
+                    new TNonEmptyList(Type::getInt()),
+                ]),
+                'tmp_name' => new Union([
+                    new TString(),
+                    new TNonEmptyList(Type::getString()),
+                ]),
+                'error' => new Union([
+                    new TInt(),
+                    new TNonEmptyList(Type::getInt()),
+                ]),
+            ];
+
+            $php_version = $config->getPhpVersionFromConfig();
```

Comment:
Please use `$codebase->analysis_php_version_id` here. I believe the version in config can be overwritten by command line for example. The codebase holds the definite version